cmd/account/iam-cred-rotate.go

```diff
@@ -0,0 +1,1325 @@
+	var nse *iamTypes.NoSuchEntityException
+	if (err != nil && errors.As(err, &nse)) || checkUser == nil {
+		o.log.Warn(o.ctx, "User Not Found: '%s', trying user:'%s' instead...\n", osdManagedAdminUsername, common.OSDManagedAdminIAM)
+		osdManagedAdminUsername = common.OSDManagedAdminIAM
+		checkUser, err := o.awsClient.GetUser(&iam.GetUserInput{UserName: awsSdk.String(osdManagedAdminUsername)})
+		if err != nil {
+			return "", err
+		}
+		if checkUser == nil {
+			return "", fmt.Errorf("user Not Found:'%s'", o.osdManagedAdminUsername)
+		}
+	}
+	if err != nil {
+		return "", err
+	}
+	return osdManagedAdminUsername, nil
+}
+
+// Cli helper to print aws access key info and return
+func (o *rotateCredOptions) cliPrintOsdManagedAdminCreds() error {
+	osdManagedAdminUsername, err := o.checkOsdManagedUsername()
+	if err != nil {
+		return err
+	}
+	currKeys, clientInput, err := o.getKeyInfo(osdManagedAdminUsername, common.AWSAccountNamespace, o.secretName)
+	if err != nil {
+		return err
+	}
+	err = o.printKeyInfo(currKeys, clientInput, osdManagedAdminUsername)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *rotateCredOptions) doRotateManagedAdminAWSCreds() error {
+	if o.awsClient == nil {
+		return fmt.Errorf("AWS client has not been created yet for doRotateAWSCreds()")
+	}
+	// Update osdManagedAdmin secrets
+	osdManagedAdminUsername, err := o.checkOsdManagedUsername()
+	if err != nil {
+		return err
+	}
+	o.log.Info(o.ctx, "Using osdManagedAdminUsername: '%s' \n", osdManagedAdminUsername)
+	// Check for max keys before attempting to create, use user provided input to delete a key if needed...
+	err = o.checkAccessKeysMaxDelete(osdManagedAdminUsername, common.AWSAccountNamespace, o.secretName, "")
+	if err != nil {
+		return err
+	}
+
+	o.log.Debug(o.ctx, "----------------------------------------------------------------------\n")
+	o.log.Debug(o.ctx, "Creating new access key for '%s'\n", osdManagedAdminUsername)
+	o.log.Debug(o.ctx, "----------------------------------------------------------------------\n")
+
+	o.log.Info(o.ctx, "Creating new access key...\n")
+	createAccessKeyOutput, err := o.awsClient.CreateAccessKey(&iam.CreateAccessKeyInput{UserName: awsSdk.String(osdManagedAdminUsername)})
+	if err != nil {
+		o.log.Warn(o.ctx, "Failed to create access key for user:'%s'\n", osdManagedAdminUsername)
+		return err
+	}
+
+	fmt.Printf("New creds:\nuser:'%s'\nAccessKey:'%s'\n", *createAccessKeyOutput.AccessKey.UserName, *createAccessKeyOutput.AccessKey.AccessKeyId)
+	// Todo: Make displaying the secret a cli provided arg? If this is useful for error/script failure recovery?
+	if o.saveSecretKeyToFile {
+		err := os.WriteFile(saveFileManaged, []byte(*createAccessKeyOutput.AccessKey.SecretAccessKey), 0600)
+		if err != nil {
+			o.log.Error(o.ctx, "Error! Saving key info to file:'%s', err:%s", saveFileManaged, err)
+			//return err
+		} else {
+			fmt.Printf("\nSaved key to: '%s'\n", saveFileManaged)
+		}
+	}
```

Comment:
Current behavior is to build and save the yaml representation of the secret in /tmp and print the path to this file(s). These can be used for apply -f or just review later. This will only happen if the key(s) have been created and something fails during the secret update().

Reply:
This has been removed. No longer saving the secret, instead there's additional checks/retry around a failed syncset allowing a user to address issues and reapply a failed syncset continuing the progress of this command.